### ISSUE TITLE
[Bug]: aria-invalid has no visual effect on RadioGroup

### ISSUE THREAD

**Issue by cpendery** [microsoft/fluentui#30921]
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
System:
    OS: Windows 11 10.0.22631
    CPU: (16) x64 11th Gen Intel(R) Core(TM) i9-11950H @ 2.60GHz
    Memory: 35.09 GB / 63.71 GB
  Browsers:
    Edge: Chromium (123.0.2420.65)
    Internet Explorer: 11.0.22621.1
```

### Are you reporting Accessibility issue?

yes

### Reproduction

https://stackblitz.com/edit/qfcdg8?file=src%2Fexample.tsx

## Actual Behavior
There is no visual indication of `aria-invalid` for the [RadioGroup](https://react.fluentui.dev/?path=/docs/components-radiogroup--default) component while there is for other form controls.

## Expected Behavior
There should be a visual indication of `aria-invalid` like making the radio buttons have red outlines, etc.

### Requested priority

High

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by cpendery** [comment]
It also has no effect on [checkbox](https://react.fluentui.dev/?path=/docs/components-checkbox--default#required)

**Comment by mainframev** [comment]
Hi @cpendery 👋🏻 

To visually indicate an error state for components like Radio, Checkbox, or RadioGroup, you can simply wrap them in the Field component and use the validationMessage prop. This approach provides a clear visual cue when there is an error.

![image](https://github.com/microsoft/fluentui/assets/14054752/db932802-dfbf-4276-a1db-d4ff3e99dd3b)

After consulting with the designers, we’ve decided not to add a separate visual indication for these components at this time. Here’s the reasoning from the designers:

> _We looked at aria-invalid and error states on other related components like input and thought about how users might encounter these problems in forms and maybe things like wizard flows. 
 
And overall, we all agreed that it felt problematic to provide options to users that might return in error--if we're providing options, the goal should be to provide valid options whenever possible. But given that errors can still happen, we wondered if providing the error message to the group of options would be better than to disrupt the specific cadence of the radio group (or checkbox, I noticed the bug mentioned that aria-invalid isn't available for checkbox either)._ 

I am closing this, as there is no further action needed 🙂
